casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -271,6 +271,56 @@ object BlockAPI {
     )
   }
 
+  private def findBlockWithDeploy[F[_]: Monad: Log: BlockStore](
+      blockHashes: Vector[BlockHash],
+      user: ByteString,
+      timestamp: Long
+  ): F[Option[BlockMessage]] =
+    blockHashes match {
+      case blockHash +: rem =>
+        for {
+          block <- ProtoUtil.unsafeGetBlock[F](blockHash)
+          result <- if (ProtoUtil.containsDeploy(block, user, timestamp)) {
+                     Option[BlockMessage](block).pure[F]
+                   } else {
+                     findBlockWithDeploy[F](rem, user, timestamp)
+                   }
+        } yield result
+      case _ => none[BlockMessage].pure[F]
```

Comment:
I prefer to explicitly mark it as `Nil`. It is easy to read and obvious that you handle empty list case. `case _ => ` can match anything.

Reply:
Yup thanks for the simplification @odeac